src/client/node/nodeKwil.ts

```diff
@@ -44,17 +48,90 @@ export class NodeKwil extends Kwil<EnvironmentType.NODE> {
       return await this.callClient(actionBody);
     }
 
-    let tempCookie: string | undefined;
+    // Ensure auth mode is set
+    await this.ensureAuthenticationMode();
+
+    // Set Temporary Cookie
+    const tempCookie = this.setTemporaryCookie(actionBody);
+
+    // Build Message
+    const message = await this.buildMessage(actionBody, kwilSigner);
+
+    // kwil.call()
+    let response = await this.callClient(message);
+
+    // Reset Cookie
+    if (this.authMode === AuthenticationMode.OPEN) {
```

Comment:
I think there is a logic issue here where you will end up resetting the cookie every time this is called.

Reply:
Let's talk live through this, but TL;DR, I think you will reset the cookie to null with every call, which will require KGW users to reauthenticate on each call.